File: tech-studio-projects/financial-industry/algo-trading-platform/cpp/orders/ModifyOrder.cpp
#include "ModifyOrder.h"

// Constructor to initialize the ModifyOrder with core order data, modification price, quantity, and buy/sell side.
ModifyOrder::ModifyOrder(std::shared_ptr<IOrderCore> orderCore, long modifyPrice, unsigned int modifyQuantity, bool isBuySide)
    : _orderCore(orderCore), _price(modifyPrice), _quantity(modifyQuantity), _isBuySide(isBuySide) {
}

// Getter for the modification price.
long ModifyOrder::GetPrice() const {
    return _price;
}

// Getter for the modification quantity.
unsigned int ModifyOrder::GetQuantity() const {
    return _quantity;
}

// Getter for the buy/sell side.
bool ModifyOrder::IsBuySide() const {
    return _isBuySide;
}

// Convert this modify order to a cancel order.
std::shared_ptr<CancelOrder> ModifyOrder::ToCancelOrder() const {
    return std::make_shared<CancelOrder>(*this);
}

// Convert this modify order to a new order.
std::shared_ptr<Order> ModifyOrder::ToNewOrder() const {
    return std::make_shared<Order>(*this);
}

// Getters from IOrderCore.
long ModifyOrder::GetOrderId() const {
    return _orderCore->GetOrderId();
}

long ModifyOrder::GetUsername() const {
    return _orderCore->GetUsername();
}

long ModifyOrder::GetSecurityId() const {
    return _orderCore->GetSecurityId();
}

File: tech-studio-projects/financial-industry/algo-trading-platform/java/algo-trading-platform/src/com/fintech/algotrading/orders/ModifyOrder.java
package com.fintech.algotrading.orders;

import java.util.Optional;

//Class representing a modified order.
public class ModifyOrder {
 private final IOrderCore orderCore;
 private final long price;
 private final int quantity;
 private final boolean isBuySide;

 // Constructor to initialize the ModifyOrder with core order data, modification price, quantity, and buy/sell side.
 public ModifyOrder(IOrderCore orderCore, long price, int quantity, boolean isBuySide) {
     this.orderCore = orderCore;
     this.price = price;
     this.quantity = quantity;
     this.isBuySide = isBuySide;
 }

 // Getter for the modification price.
 public long getPrice() {
     return price;
 }

 // Getter for the modification quantity.
 public int getQuantity() {
     return quantity;
 }

 // Getter for the buy/sell side.
 public boolean isBuySide() {
     return isBuySide;
 }

 // Convert this modify order to a cancel order.
 public CancelOrder toCancelOrder() {
     return new CancelOrder();
 }

 // Convert this modify order to a new order.
 public Order toNewOrder() {
     return new Order(price, null, price, quantity, isBuySide);
 }

 // Getters from IOrderCore.
 public long getOrderId() {
     return orderCore.getOrderId();
 }

 public long getUsername() {
     return orderCore.getUsername();
 }

 public long getSecurityId() {
     return orderCore.getSecurityId();
 }
}
